plugins/cue_points/quiz/lib/model/QuizUserEntry.php

```diff
@@ -23,39 +23,53 @@ public function setScore($v){ $this->putInCustomData("score", $v);}
 	public function getScore(){ return $this->getFromCustomData("score");}
 
 	/**
-	 * @param $entryId
 	 * @return int
 	 */
 	public function calculateScore()
 	{
 		$finalScore = 0;
-		$answerType = QuizPlugin::getCuePointTypeCoreValue(QuizCuePointType::QUIZ_ANSWER);
-		$answers = CuePointPeer::retrieveByEntryId($this->getEntryId(), array($answerType));
-		foreach ($answers as $answer)
+		$questionType = QuizPlugin::getCuePointTypeCoreValue(QuizCuePointType::QUIZ_QUESTION);
+		$questions = CuePointPeer::retrieveByEntryId($this->getEntryId(), array($questionType));
+		$totalPoints = 0;
+		$userPoints = 0;
+		foreach ($questions as $question)
 		{
 			/**
-			 * @var AnswerCuePoint $answer
-			 */
-			$question = CuePointPeer::retrieveByPK($answer->getParentId());
-			/**
 			 * @var QuestionCuePoint $question
 			 */
+			$answersCrit = new Criteria();
+			$answersCrit->add(CuePointPeer::ENTRY_ID,$this->getEntryId());
+			$answersCrit->add(CuePointPeer::PARENT_ID, $question->getId());
+			$answers = CuePointPeer::doSelect($answersCrit);
+			$currAnswer = null;
+			foreach ($answers as $answer)
+			{
+				/**
+				 * @var AnswerCuePoint $answer
+				 */
+				if ($answer->getQuizUserEntryId() == $this->id)
```

Comment:
Can't the resulted $answers array be huge? Seems like the wrong way to filter it.